src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -731,8 +748,91 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
-    private void writeServerEnvProperties(File file, Properties properties, Properties projectProperties) throws IOException {
+    private void setServerEnv(File envFile, String serverEnvPath) {
+        Properties configuredProps = combineServerEnvProperties(server.env, envProjectProps);
+
+        if(server.appendServerEnv) {
+            setServerEnvWithAppendServerEnvHelper(envFile, serverEnvPath, configuredProps)
+        }
+        else {
+            setServerEnvHelper(envFile, serverEnvPath, configuredProps)
+        }
+    }
+
+    private void setServerEnvWithAppendServerEnvHelper(File envFile, String serverEnvPath, Properties configuredProps) {
+        Properties serverEnvProps = convertServerEnvToProperties(envFile);
+        Properties mergedProperties = new Properties();
+
+        if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
+            if (serverEnvPath != null) {
+                logger.warn("The " + serverEnvPath + " file is merged with inlined configuration.")
+            }
+
+            //Create properties from existing server env
+            //Merge server.env props with generated serverEnvPath
+            mergedProperties = combineServerEnvProperties(serverEnvProps, configuredProps);
+            // writeServerEnvProperties(envFile, mergedProperties);
+        }
+
+        if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
```

Comment:
This is merging the serverEnvFile last which is not correct. The project properties and server.env take precedence. You need to do this in the following order.

1. Combine the serverEnvFile properties with the envFile properties (serverEnvProps in your method).
2. Combine the configuredProps (which are the envProjectProps combined with the server.env properties) with the combined properties from step 1.